src/base/utils/looks_like.jl

```diff
@@ -97,6 +98,66 @@ forwarded to [`looks_like_biomass_reaction`](@ref).
 find_biomass_reaction_ids(m::MetabolicModel; kwargs...) =
     filter(id -> looks_like_biomass_reaction(id, kwargs...), reactions(m))
 
+"""
+    looks_like_internal_reaction(
+        rxn_id::String;
+        exchange_prefixes = _constants.exchange_prefixes,
+        biomass_strings = _constants.biomass_strings,
+    )::Bool
+
+A predicate that matches reaction identifiers that look like
+internal reactions, i.e. reactions that are neither exchange nor biomass reactions.
+Exchange reactions are identified based on matching prefixes in
+the set `exchange_prefixes` and biomass reactions are identified by looking for
+occurences of `biomass_strings` in the reaction id.
+
+Also see [`find_internal_reactions`](@ref).
+
+# Example
+```
+findall(looks_like_internal_reaction, reactions(model)) # returns indices
+filter(looks_like_internal_reaction, reactions(model)) # returns Strings
+
+# to use the optional arguments you need to expand the function's arguments
+# using an anonymous function
+findall(x -> looks_like_internal_reaction(x; exchange_prefixes=["EX_", "R_EX_"]), reactions(model)) # returns indices
+filter(x -> looks_like_internal_reaction(x; exchange_prefixes=["EX_", "R_EX_"]), reactions(model)) # returns Strings
+```
+"""
+function looks_like_internal_reaction(
+    rxn_id::String;
+    exchange_prefixes = _constants.exchange_prefixes,
+    biomass_strings = _constants.biomass_strings,
+)::Bool
+    !looks_like_biomass_reaction(
+        rxn_id;
+        exchange_prefixes = exchange_prefixes,
+        biomass_strings = biomass_strings,
+    ) &&
+        !looks_like_exchange_reaction(
+            rxn_id;
+            exchange_prefixes = exchange_prefixes,
+            biomass_strings = biomass_strings,
+        )
+end
+
+"""
+    find_internal_reactions(m::MetabolicModel; kwargs...)
+
+Shortcut for finding internal reaction indices in a model; arguments are
+forwarded to [`looks_like_internal_reaction`](@ref).
+"""
+find_internal_reactions(m::MetabolicModel; kwargs...) =
+    findall(id -> looks_like_internal_reaction(id; kwargs...), reactions(m))
+
+"""
+    find_internal_reaction_ids(m::MetabolicModel; kwargs...)
+
+Shortcut for finding internal reaction identifiers in a model; arguments are
+forwarded to [`looks_like_internal_reaction`](@ref).
+"""
+find_internal_reaction_ids(m::MetabolicModel; kwargs...) =
+    filter(id -> looks_like_internal_reaction(id, kwargs...), reactions(m))
```

Comment:
we should eventually generalize this to something like

`find_reactions(looks_like_internal_reaction, model)`

Reply:
anyway the indexish variant sounds like it's more general while it's in fact not, maybe name the stuff as `find_internal_reaction_indexes` vs `find_internal_reaction_ids`

Reply:
So the other `find...` functions look like this, this would be changed in another PR since I would then need to change all the other ones to make things homogeneous